### PR TITLE
Add Structured Output Support for ToolCallLLM

### PR DESCRIPTION
**Pull request by Aryan-Railtown**
## Description
This PR adds the ability to return structured output from ToolCallLLM by implementing the `StructuredToolCallLLM` class and extending the `tool_call_llm` function to accept Pydantic models as `output_model`.

### Added
- Implemented `StructuredToolCallLLM` class that allows returning structured Pydantic models from tool call chains
- Added support for passing Pydantic models to the `tool_call_llm` function's `output_model` parameter
- Created comprehensive test suite for structured output functionality

### Technical Details
- `StructuredToolCallLLM` uses a dedicated structured response node to convert conversation history into the specified Pydantic model
- The implementation handles empty models, nested models, and complex field types
- The easy-usage wrapper (`tool_call_llm`) now accepts both string literals for `output_types` and Pydantic models as `output_model`

### Changes
```python
# Before
def tool_call_llm(
    ...,
    output_type: Literal["MessageHistory", "LastMessage"] = "LastMessage",
):
    ...

# After
def tool_call_llm(
    ...,
    output_type: Optional[Literal["MessageHistory", "LastMessage"]] = "LastMessage",
    output_model: Optional[Type[BaseModel]] = None,
):
    ...
```

### Example Usage
```python
# Define a Pydantic model for structured output
class MathOutput(BaseModel):
    sum: float = Field(description="The sum of the random numbers")
    random_numbers: List[int] = Field(description="The list of random numbers generated")

# Create a node with structured output
math_node = rc.library.tool_call_llm(
    connected_nodes={rng_node},
    pretty_name="Math Node", 
    system_message=system_message,
    model=model,
    output_model=MathOutput,
)

# Use the node
with rc.Runner() as runner:
    response = await runner.run(math_node, message_history=message_history)
    # Access structured data
    total = response.answer.sum
    numbers = response.answer.random_numbers
```
## Additional
- Restructured wrapper functions into an easy_usage_wrappers folder for better organization
- Maintained backward compatibility via updated `__init__.py`

## Testing
- Added test cases for basic functionality, empty models, complex nested models, and edge cases
- Ensured backward compatibility with existing code using "MessageHistory" and "LastMessage" output types
- All existing tests pass with the new changes

---
## Checklist for Author

- [x] Linked your tickets to this pull request
- [x] Ran tests locally and all pass

---

### CODE REVIEW

File: tests/rc_tests/nodes/library/test_terminal_llm.py [hunk 1]
```diff
@@ -31,25 +31,7 @@ async def test_call_capitalize_text_stream():
...
-def decoder_system_message():
-    return rc.llm.SystemMessage("You are a text decoder. Decode the bytes into a string.")
-
-
-@pytest.fixture
-def model():
-    return rc.llm.OpenAILLM("gpt-4o")
-
-
+# ======== Tests using TerminalLLMs as tools ==========
```
Comment (Aryan-Railtown) [inline tests/rc_tests/nodes/library/test_terminal_llm.py:34]: moved these fixtures to conftest.py

File: tests/rc_tests/nodes/library/utils.py
Comment (Aryan-Railtown) [inline tests/rc_tests/nodes/library/utils.py:1]: moved the helper function in this file to conftest.py

File: tests/rc_tests/nodes/library/test_function.py
Comment (Aryan-Railtown) [inline tests/rc_tests/nodes/library/test_function.py:1]: using create_top_level_node as a fixture directly from conftest.py

File: src/requestcompletion/nodes/library/structured_tool_call_llm.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    def __init__(
+        self,
+        message_history: MessageHistory,
+        llm_model: ModelBase,
+        output_model: BaseModel,
+    ):
+
+        super().__init__(message_history, llm_model)
+        system_structured = SystemMessage(
```
Comment (soulFood5632) [inline src/requestcompletion/nodes/library/structured_tool_call_llm.py:19]: We may want to do some testing here to show that this prompt is the best.

File: src/requestcompletion/nodes/library/structured_tool_call_llm.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            "You are a structured LLM that can convert the response into a structured output."
+        )
+        self.structured_resp_node = structured_llm(output_model, system_message=system_structured, model=self.model)
+
+    async def return_output(self) -> BaseModel:
+        last_message = self.message_hist[-1]
+        try:
```
Comment (soulFood5632) [inline src/requestcompletion/nodes/library/structured_tool_call_llm.py]: From a design perspective I think it would make more sense to move this logic into the `invoke` method. I know it's kinda an awkward difference, but I think it best for the return_output method to be a simple as possible accessing the saved state and returning and output. Leaving all the more complicated details to the other function.

File: src/requestcompletion/nodes/library/tool_call_llm.py [hunk 4]
```diff
@@ -5,6 +5,6 @@
 
 class ToolCallLLM(OutputLessToolCallLLM[str], ABC):
 
-    def return_output(self):
+    async def return_output(self):
```
Comment (soulFood5632) [inline src/requestcompletion/nodes/library/tool_call_llm.py]: Related to my other comment but I think it best to make this thing not async, and keep things simple.

File: src/requestcompletion/nodes/library/structured_tool_call_llm.py (hunk 3 above)
Comment (Aryan-Railtown) [reply src/requestcompletion/nodes/library/structured_tool_call_llm.py]: that makes sense. I'll move the logic into invoke.

File: src/requestcompletion/nodes/library/_tool_call_llm_base.py [hunk 5]
```diff
@@ -91,63 +85,16 @@ async def invoke(
                 # the message is malformed from the model
                 raise RuntimeError("ModelLLM returned an unexpected message type.",
                 )
-
+        
+        if self.structured_resp_node:
+            last_message = self.message_hist[-1]
```
Comment (Amir-R25) [inline src/requestcompletion/nodes/library/_tool_call_llm_base.py:90]: Are we sure we only need the last message to get the structured response?

Comment (Aryan-Railtown) [reply src/requestcompletion/nodes/library/_tool_call_llm_base.py:90]: Yes, I made sure the last message was enough. In case of the tool_call_llm, the last assistant message contains all the necessary info to make a structured resp. I tried out message_hist as well as last message only, both seem to have the same behavior.
